From 00fdbb9a48fa1aadcb98ff7246fc098f480e84e1 Mon Sep 17 00:00:00 2001
From: Laura Lawrence <Laura.Lawrence@freescale.com>
Date: Tue, 10 Jun 2008 15:51:04 -0500
Subject: [PATCH] ENGR00079741 Fix SSI Interrupt Enable Problem

SSI SIER not initialized properly
SSI interrupts enabled by default at reset must be disabled,
else interrupts will occur constantly, and audio wont work.
Bug introduced in ENGR00077952 patch

Signed-off-by: Laura Lawrence <Laura.Lawrence@freescale.com>
---
 sound/soc/imx/imx-ssi.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/sound/soc/imx/imx-ssi.c b/sound/soc/imx/imx-ssi.c
index c391cb4..16177e2 100644
--- a/sound/soc/imx/imx-ssi.c
+++ b/sound/soc/imx/imx-ssi.c
@@ -338,6 +338,7 @@ static int imx_ssi_startup(struct snd_pcm_substream *substream)
 		    SSI_SFCSR_RFWM0(SSI_RXFIFO_WATERMARK) |
 		    SSI_SFCSR_TFWM1(SSI_TXFIFO_WATERMARK) |
 		    SSI_SFCSR_TFWM0(SSI_TXFIFO_WATERMARK);
+		SSI1_SIER = 0;
 	} else {
 
 		if (ssi_active[SSI2_PORT]++)
@@ -352,6 +353,7 @@ static int imx_ssi_startup(struct snd_pcm_substream *substream)
 		    SSI_SFCSR_RFWM0(SSI_RXFIFO_WATERMARK) |
 		    SSI_SFCSR_TFWM1(SSI_TXFIFO_WATERMARK) |
 		    SSI_SFCSR_TFWM0(SSI_TXFIFO_WATERMARK);
+		SSI2_SIER = 0;
 	}
 
 	SSI_DUMP();
-- 
1.5.4.4

